cfn-resources/custom-dns-configuration-cluster-aws/cmd/resource/resource.go

```diff
@@ -0,0 +1,165 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{constants.ProjectID, constants.PubKey, constants.PvtKey}
+var ReadRequiredFields = []string{constants.ProjectID, constants.PubKey, constants.PvtKey}
+var UpdateRequiredFields []string
```

Comment:
use one array variable if all the input fields are same for create,read,delete,list